### PR TITLE
Timer

### PR DESCRIPTION
**Pull request by yhj050224**
Made it so that timer runs in studysessionview(with start and stop buttons). Also created new view called breaksessionview. Allowed transition from study -> break when timer reaches 0. Allowed transition from study -> set up when button is clicked.

### CODE REVIEW

File: src/main/java/view/BreakSessionView.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import javax.swing.SwingConstants;
+import javax.swing.Timer;
+
+import constants.Constants;
+import interface_adapter.break_session.BreakSessionController;
+import interface_adapter.break_session.BreakSessionViewModel;
+import interface_adapter.logout.LogoutController;
+import interface_adapter.study_session.StudySessionController;
+import use_case.breaksession.BreakSessionOutputBoundary;
+import use_case.studysession.StudySessionOutputBoundary;
```
Comment (chiually) [inline src/main/java/view/BreakSessionView.java:24]: checkstyle: unused imports

File: src/main/java/app/builder/view/SessionViewBuilder.java [hunk 2]
```diff
@@ -85,14 +87,19 @@ public SessionViewBuilder buildSetupSessionView() {
...
-                viewModels.getTimerViewModel(),
-                catViewsAndModels.getViewModels().getCatViewModel(),
-                catViewsAndModels.getViewModels().getDisplayCatStatsViewModel(),
+                dialogService,
                 catViewsAndModels.getViews().getCatView());
         cardPanel.add(studySessionView, studySessionView.getViewName());
         return this;
     }
 
+    public SessionViewBuilder buildBreakSessionView() {
```
Comment (chiually) [inline src/main/java/app/builder/view/SessionViewBuilder.java:96]: javadoc

File: src/main/java/view/SetupSessionView.java [hunk 3]
```diff
@@ -18,11 +18,10 @@
  * The view for when the user is setting up their study session.
  */
 public class SetupSessionView extends JPanel implements ActionListener {
-    private final String viewName = "setup study session";
     private final SetupSessionViewModel setupSessionViewModel;
-    private String[] studyChoices = {"10 mins", "15 mins", "20 mins", "25 mins", "30 mins", "35 mins", "40 mins",
-        "45 mins", "50 mins", "55 mins", "60 mins"};
-    private String[] breakChoices = {"5 mins", "10 mins"};
+    private String[] studyChoices = {"10 mins", "15 mins", "20 mins", "25 mins", "30 mins", "35 mins",
+            "40 mins", "45 mins", "50 mins", "55 mins", "60 mins" };
```
Comment (chiually) [inline src/main/java/view/SetupSessionView.java:23]: checkstyle: indentation of array

File: src/main/java/app/factory/ViewFactory.java [hunk 4]
```diff
@@ -173,13 +170,7 @@ public DisplayCatStatsView createDisplayCatStatsView(JFrame parent,
...
     }
 
-    /**
-     * Creates a Timer View.
-     * @param timerViewModel the timer view model
-     * @return TimerView
-     */
-    public TimerView createTimerView(TimerViewModel timerViewModel) {
-        return new TimerView(timerViewModel);
+    public BreakSessionView createBreakSessionView(BreakSessionViewModel breakSessionViewModel) {
```
Comment (chiually) [inline src/main/java/app/factory/ViewFactory.java:173]: add a javadoc

File: src/main/java/app/factory/ViewFactory.java [hunk 5]
```diff
@@ -112,17 +112,14 @@ public SetupSessionView createSetupSessionView(SetupSessionViewModel setupSessio
     /**
      * Creates a Study Session View.
      * @param studySessionViewModel the study session view model
-     * @param timerViewModel the timer view model
      * @param catViewModel the cat view model
```
Comment (chiually) [inline src/main/java/app/factory/ViewFactory.java:115]: parameters in the javadoc don't match the ones in the method

File: src/main/java/app/builder/usecase/TimerUseCaseBuilder.java [hunk 6]
```diff
@@ -2,16 +2,23 @@
 
 import app.builder.view.Views;
 import app.components.DataAccessComponents;
+import interface_adapter.break_session.BreakSessionController;
+import interface_adapter.break_session.BreakSessionPresenter;
 import interface_adapter.setupsession.SetupSessionController;
 import interface_adapter.setupsession.SetupSessionPresenter;
+import interface_adapter.study_session.StudySessionController;
+import interface_adapter.study_session.StudySessionPresenter;
 import interface_adapter.timer.TimerController;
```
Comment (chiually) [inline src/main/java/app/builder/usecase/TimerUseCaseBuilder.java:11]: unused imports

File: src/main/java/interface_adapter/break_session/BreakSessionPresenter.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    public BreakSessionPresenter(ViewManagerModel viewManagerModel, LoginViewModel loginViewModel,
+                                 StudySessionViewModel studySessionViewModel) {
+        this.viewManagerModel = viewManagerModel;
+        this.loginViewModel = loginViewModel;
+        this.studySessionViewModel = studySessionViewModel;
+    }
+
+    @Override
+    public void switchToStudySessionView() {
+        System.out.println("Switching to Study Session View...");
```
Comment (chiually) [inline src/main/java/interface_adapter/break_session/BreakSessionPresenter.java:27]: do you want to keep the print statement?

File: src/main/java/view/StudySessionView.java [hunk 8]
```diff
@@ -9,62 +9,98 @@
...
+                    updateTimerLabel();
+                }
+                else {
+                    swingTimer.stop();
+                    System.out.println("Time is up! Switching to Break Session...");
+
+                    // Notify the presenter or controller to switch the view
+                    if (studySessionController != null) {
+                        System.out.println("test");
+                        studySessionController.switchToBreakSessionView();
```
Comment (chiually) [inline src/main/java/view/StudySessionView.java:95]: remove test print statement if done working on it

**Review by chiually — COMMENTED** [review]
good work just fix up the javadocs, remove print statements if you don't need them anymore, and remove unused imports

**Review by jehapark — APPROVED** [review]
Good work. Some checkstyle errors left to be fixed.

**Pull request merged**